ansible_playbook.yml

```diff
@@ -0,0 +1,90 @@
+-
+  hosts: all
+  remote_user: vagrant
+  sudo: yes
+  tasks: 
+    - name: "update"
+      apt:
+        update_cache: yes
+    - name: "Install linux-image-extra-"
+      apt:
+          name: linux-image-extra-4.13.0-36-generic
+          state: latest
+    - name: 'Install linux-image-extra-'
+      apt:
+        name: linux-image-extra-virtual
+        state: latest
+    - name: 'Install apt-transport-https'
+      apt:
+        name: apt-transport-https
+        state: latest
+    - name: 'Install ca-certificates'
+      apt:
+        name: ca-certificates
+        state: latest
+    - name: 'Install software-properties-common'
+      apt:
+        name: software-properties-common
+        state: latest
+    - name: 'Add Docker apt key'
+      apt_key:
+        url: 'https://download.docker.com/linux/ubuntu/gpg'
+        id: 9DC858229FC7DD38854AE2D88D81803C0EBFCD88
+        state: present
+      register: add_repository_key
+      ignore_errors: true
```

Comment:
Why do we ignore errors here?

Reply:
I think there's no need to do 'dry-run' for adding apt key. I would suggest skipping this step, ensure curl is present unconditionally and then download the key - if curl is installed nothing will happen and we want to fail anyway if the key won't get downloaded properly.